Sources/XCTest/Public/XCTAssert.swift

```diff
@@ -421,3 +423,14 @@ public func XCTAssertThrowsError<T>(_ expression: @autoclosure () throws -> T, _
         }
     }
 }
+
+public func XCTAssertNoThrow<T>(_ expression: @autoclosure () throws -> T, _ message: @autoclosure () -> String = "", file: StaticString = #file, line: UInt = #line) {
+    _XCTEvaluateAssertion(.noThrow, message: message, file: file, line: line) {
+        do {
+             _ = try expression()
+            return .success
+        } catch _ {
+            return .expectedFailure("\(filename):\(line): XCTAssertNoThrow failed: threw error \"\(error)\". : \(message)")
```

Comment:
Note that a lot of the pieces of this failure message are produced in the `_XCTEvaluateAssertion` function and won't need to be included on this line here. I think you'll want to end up with with an expression like:

```swift
    return .expectedFailure("threw error \"\(error)\"")
```